cognite/powerops/resync/v2/shop_to_assets.py

```diff
@@ -0,0 +1,349 @@
+
+        all_connections = shop_model["connections"]
+        all_junctions = shop_model["model"].get("junction", {})
+        all_tunnels = shop_model["model"].get("tunnel", {})
+        inlet_reservoir_name, connection_losses = self._plant_to_inlet_reservoir_with_losses(
+            name, all_connections, all_junctions, all_tunnels, set(reservoir_by_name.keys())
+        )
+
+        plant_generators: list[GeneratorWrite] = []
+        for connection in all_connections:
+            if (
+                connection.get("from_type") == "plant"
+                and connection["from"] == name
+                and (gen := generator_by_name.get(connection["to"]))
+            ):
+                plant_generators.append(gen)
+            elif (
+                connection.get("to_type") == "plant"
+                and connection["to"] == name
+                and (gen := generator_by_name.get(connection["from"]))
+            ):
+                plant_generators.append(gen)
+
+        return PlantWrite(
+            external_id=f"plant_{name}",
+            name=name,
+            display_name=display_name,
+            outlet_level=float(data.get("outlet_line", 0)),
+            p_min=float(data.get("p_min", self.p_min_fallback)),
+            p_max=float(data.get("p_max", self.p_max_fallback)),
+            ordering=order,
+            penstock_head_loss_factors={
+                str(penstock_number): float(loss_factor)
+                for penstock_number, loss_factor in enumerate(
+                    data.get("penstock_loss", [self.head_loss_factor_fallback]), start=1
+                )
+            },
+            head_loss_factor=float(data.get("main_loss", [self.head_loss_factor_fallback])[0]),
+            connection_losses=connection_losses,
+            water_value_time_series=plant_timeseries.get("water_value"),
+            inlet_level_time_series=plant_timeseries.get("inlet_reservoir_level"),
+            outlet_level_time_series=plant_timeseries.get("outlet_reservoir_level"),
+            p_min_time_series=plant_timeseries.get("p_min"),
+            p_max_time_series=plant_timeseries.get("p_max"),
+            feeding_fee_time_series=plant_timeseries.get("feeding_fee"),
+            head_direct_time_series=plant_timeseries.get("head_direct"),
+            watercourse=watercourse_xid,
+            inlet_reservoir=reservoir_by_name.get(inlet_reservoir_name),
+            generators=plant_generators,
+        )
+
+    def _to_reservoir(self, name: str, _: dict, reservoir_display_name_and_order: dict) -> ReservoirWrite:
+        display_name, order = reservoir_display_name_and_order.get(name, (name, 999))
+
+        return ReservoirWrite(external_id=f"reservoir_{name}", name=name, display_name=display_name, ordering=order)
+
+    def _plant_to_inlet_reservoir_with_losses(
+        self, plant_name: str, all_connections: list[dict], all_junctions: dict, all_tunnels: dict, reservoirs: set[str]
+    ) -> tuple[str, float]:
```

Comment:
This is a copy-paste of the previous function doing the same work.

Reply:
I don't necessarily agree that this shouldn't be reviewed now, especially since there's no test coverage for it, but I've skipped it for now as there are other things requiring updates.

Reply:
To do this properly requires an effort similar to this PR in size, so lets keep this as a technical debt due to the urgency of getting resync up and running with the new models (it is currently a blocker for some of the function refactoring).

Reply:
As a minimum put a `TODO` comment so that there is actually a possibility it will be picked up at a later date

Reply:
Todo's have strong tendency to be lost, so tracking this with: https://cognitedata.atlassian.net/browse/POWEROPS-2224?atlOrigin=eyJpIjoiNGFhYWQxMGU0NTE3NGEzNDlhZTBkN2Y5NDhkYTczYmYiLCJwIjoiaiJ9

Reply:
Jira tickets also get lost 😉 but I don't really have a preference as long as it's documented somewhere to be picked up later

Reply:
Yes, but Jira serves as the one place to look. Currently, we have multiple code bases, so harder to keep in track of todos in them.